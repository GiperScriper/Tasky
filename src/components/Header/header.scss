@import 'assets/styles/_vars';

.header {
  display: flex;
  justify-content: space-between;
  align-items: center;

  padding: 10px;

  box-shadow: 0px 0px 14px -5px $text-color;
  &__logo {
    margin-right: 50px;

    font-size: 35px;
    line-height: 35px;
    color: $blue;
  }
  &__menu {
    display: flex;
    flex-grow: 1;

    margin: 0;
    padding: 0;

    list-style: none;
  }
  &__menu-link {
    margin-right: 15px;

    @include transition-animation;
    &:hover {
      color: #B6BFCC;
    }
  }
  &__notice {
    display: flex;

    font-size: 20px;
  }
  &__notice-button {
    position: relative;

    padding: 0 15px;

    color: $gray;

    border-right: 1px solid $light-gray;
    &:hover {
      color: $text-color;
    }
  }
  &__notice-indicator {
    width: 5px;
    height: 5px;

    display: block;

    position: absolute;
    right: 7px;
    top: 0;

    background-color: $blue;

    border-radius: 50%;
  }
  &__user {
    display: flex;
    align-items: center;
  }
  &__user-image {
    width: 40px;
    height: 40px;

    margin: 0 10px;

    border-radius: 50%;
  }
}