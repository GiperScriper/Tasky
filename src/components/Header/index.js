import React from 'react';
import { NavLink } from 'react-router-dom';
import './header.scss';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEarlybirds } from '@fortawesome/free-brands-svg-icons';
import { faBell, faCog, faQuestionCircle } from '@fortawesome/free-solid-svg-icons';
import userImage from '@/assets/images/user.png';

const Header = () => {
  console.log('render Header');
  return (
    <header className="header">
      <div className="header__logo">
        <FontAwesomeIcon icon={faEarlybirds} />
      </div>
      <ul className="header__menu">
        <li>
          <NavLink exact className="header__menu-link" activeClassName="header__menu-link--active" to="/">
            Dashboard
          </NavLink>
        </li>
        <li>
          <NavLink className="header__menu-link" activeClassName="header__menu-link--active" to="/projects">
            Projects
          </NavLink>
        </li>
      </ul>
      <div className="header__notice">
        <button className="button button--transparent header__notice-button">
          <FontAwesomeIcon icon={faBell} />
          <span className="header__notice-indicator" />
        </button>
        <button className="button button--transparent header__notice-button">
          <FontAwesomeIcon icon={faQuestionCircle} />
          <span className="header__notice-indicator" />
        </button>
        <button className="button button--transparent header__notice-button">
          <FontAwesomeIcon icon={faCog} />
        </button>
      </div>
      <div className="header__user">
        <img src={userImage} className="header__user-image" alt="User" />
      </div>
    </header>
  );
};

export default Header;
