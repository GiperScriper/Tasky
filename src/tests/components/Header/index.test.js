import React from 'react';
import { shallow } from 'enzyme';
import Header from '@/components/Header';

describe('<Header />', () => {
  it('should render Header component correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('NavLink').length).toBe(2);
  });
});
