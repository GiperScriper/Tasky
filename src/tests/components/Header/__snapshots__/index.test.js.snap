// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Header component correctly 1`] = `
<header
  className="header"
>
  <div
    className="header__logo"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            480,
            512,
            Array [],
            "f39a",
            "M313.2 47.5c1.2-13 21.3-14 36.6-8.7.9.3 26.2 9.7 19 15.2-27.9-7.4-56.4 18.2-55.6-6.5zm-201 6.9c30.7-8.1 62 20 61.1-7.1-1.3-14.2-23.4-15.3-40.2-9.6-1 .3-28.7 10.5-20.9 16.7zM319.4 160c-8.8 0-16 7.2-16 16s7.2 16 16 16 16-7.2 16-16-7.2-16-16-16zm-159.7 0c-8.8 0-16 7.2-16 16s7.2 16 16 16 16-7.2 16-16-7.2-16-16-16zm318.5 163.2c-9.9 24-40.7 11-63.9-1.2-13.5 69.1-58.1 111.4-126.3 124.2.3.9-2-.1 24 1 33.6 1.4 63.8-3.1 97.4-8-19.8-13.8-11.4-37.1-9.8-38.1 1.4-.9 14.7 1.7 21.6 11.5 8.6-12.5 28.4-14.8 30.2-13.6 1.6 1.1 6.6 20.9-6.9 34.6 4.7-.9 8.2-1.6 9.8-2.1 2.6-.8 17.7 11.3 3.1 13.3-14.3 2.3-22.6 5.1-47.1 10.8-45.9 10.7-85.9 11.8-117.7 12.8l1 11.6c3.8 18.1-23.4 24.3-27.6 6.2.8 17.9-27.1 21.8-28.4-1l-.5 5.3c-.7 18.4-28.4 17.9-28.3-.6-7.5 13.5-28.1 6.8-26.4-8.5l1.2-12.4c-36.7.9-59.7 3.1-61.8 3.1-20.9 0-20.9-31.6 0-31.6 2.4 0 27.7 1.3 63.2 2.8-61.1-15.5-103.7-55-114.9-118.2-25 12.8-57.5 26.8-68.2.8-10.5-25.4 21.5-42.6 66.8-73.4.7-6.6 1.6-13.3 2.7-19.8-14.4-19.6-11.6-36.3-16.1-60.4-16.8 2.4-23.2-9.1-23.6-23.1.3-7.3 2.1-14.9 2.4-15.4 1.1-1.8 10.1-2 12.7-2.6 6-31.7 50.6-33.2 90.9-34.5 19.7-21.8 45.2-41.5 80.9-48.3C203.3 29 215.2 8.5 216.2 8c1.7-.8 21.2 4.3 26.3 23.2 5.2-8.8 18.3-11.4 19.6-10.7 1.1.6 6.4 15-4.9 25.9 40.3 3.5 72.2 24.7 96 50.7 36.1 1.5 71.8 5.9 77.1 34 2.7.6 11.6.8 12.7 2.6.3.5 2.1 8.1 2.4 15.4-.5 13.9-6.8 25.4-23.6 23.1-3.2 17.3-2.7 32.9-8.7 47.7 2.4 11.7 4 23.8 4.8 36.4 37 25.4 70.3 42.5 60.3 66.9zM207.4 159.9c.9-44-37.9-42.2-78.6-40.3-21.7 1-38.9 1.9-45.5 13.9-11.4 20.9 5.9 92.9 23.2 101.2 9.8 4.7 73.4 7.9 86.3-7.1 8.2-9.4 15-49.4 14.6-67.7zm52 58.3c-4.3-12.4-6-30.1-15.3-32.7-2-.5-9-.5-11 0-10 2.8-10.8 22.1-17 37.2 15.4 0 19.3 9.7 23.7 9.7 4.3 0 6.3-11.3 19.6-14.2zm135.7-84.7c-6.6-12.1-24.8-12.9-46.5-13.9-40.2-1.9-78.2-3.8-77.3 40.3-.5 18.3 5 58.3 13.2 67.8 13 14.9 76.6 11.8 86.3 7.1 15.8-7.6 36.5-78.9 24.3-101.3z",
          ],
          "iconName": "earlybirds",
          "prefix": "fab",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
  </div>
  <ul
    className="header__menu"
  >
    <li>
      <NavLink
        activeClassName="header__menu-link--active"
        ariaCurrent="true"
        className="header__menu-link"
        exact={true}
        to="/"
      >
        Dashboard
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="header__menu-link--active"
        ariaCurrent="true"
        className="header__menu-link"
        to="/projects"
      >
        Projects
      </NavLink>
    </li>
  </ul>
  <div
    className="header__notice"
  >
    <button
      className="button button--transparent header__notice-button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f0f3",
              "M433.884 366.059C411.634 343.809 384 316.118 384 208c0-79.394-57.831-145.269-133.663-157.83A31.845 31.845 0 0 0 256 32c0-17.673-14.327-32-32-32s-32 14.327-32 32c0 6.75 2.095 13.008 5.663 18.17C121.831 62.731 64 128.606 64 208c0 108.118-27.643 135.809-49.893 158.059C-16.042 396.208 5.325 448 48.048 448H160c0 35.346 28.654 64 64 64s64-28.654 64-64h111.943c42.638 0 64.151-51.731 33.941-81.941zM224 472a8 8 0 0 1 0 16c-22.056 0-40-17.944-40-40h16c0 13.234 10.766 24 24 24z",
            ],
            "iconName": "bell",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
      <span
        className="header__notice-indicator"
      />
    </button>
    <button
      className="button button--transparent header__notice-button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f059",
              "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z",
            ],
            "iconName": "question-circle",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
      <span
        className="header__notice-indicator"
      />
    </button>
    <button
      className="button button--transparent header__notice-button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f013",
              "M444.788 291.1l42.616 24.599c4.867 2.809 7.126 8.618 5.459 13.985-11.07 35.642-29.97 67.842-54.689 94.586a12.016 12.016 0 0 1-14.832 2.254l-42.584-24.595a191.577 191.577 0 0 1-60.759 35.13v49.182a12.01 12.01 0 0 1-9.377 11.718c-34.956 7.85-72.499 8.256-109.219.007-5.49-1.233-9.403-6.096-9.403-11.723v-49.184a191.555 191.555 0 0 1-60.759-35.13l-42.584 24.595a12.016 12.016 0 0 1-14.832-2.254c-24.718-26.744-43.619-58.944-54.689-94.586-1.667-5.366.592-11.175 5.459-13.985L67.212 291.1a193.48 193.48 0 0 1 0-70.199l-42.616-24.599c-4.867-2.809-7.126-8.618-5.459-13.985 11.07-35.642 29.97-67.842 54.689-94.586a12.016 12.016 0 0 1 14.832-2.254l42.584 24.595a191.577 191.577 0 0 1 60.759-35.13V25.759a12.01 12.01 0 0 1 9.377-11.718c34.956-7.85 72.499-8.256 109.219-.007 5.49 1.233 9.403 6.096 9.403 11.723v49.184a191.555 191.555 0 0 1 60.759 35.13l42.584-24.595a12.016 12.016 0 0 1 14.832 2.254c24.718 26.744 43.619 58.944 54.689 94.586 1.667 5.366-.592 11.175-5.459 13.985L444.788 220.9a193.485 193.485 0 0 1 0 70.2zM336 256c0-44.112-35.888-80-80-80s-80 35.888-80 80 35.888 80 80 80 80-35.888 80-80z",
            ],
            "iconName": "cog",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
    </button>
  </div>
  <div
    className="header__user"
  >
    <img
      alt="User"
      className="header__user-image"
      src="user.png"
    />
  </div>
</header>
`;
